server/controllers/boardController.js

```diff
@@ -0,0 +1,33 @@
+import Board from '../models/boardModel';
+import { buildResponse } from '../utils/helpers';
+import {} from '../models/issuesModel';
+
+const getBoardDetails = async (req, res) => {
+  const id = parseInt(req.params.id, 10);
+  if (typeof id !== 'number') {
+    res.status(400).send(buildResponse(false, `${req.params.id} should be a number`));
+  }
+  try {
+    let [board] = await Board.find({ id });
+    if (!board) {
+      res.status(400).send(buildResponse(false, `No board found with id :${id}`));
+    }
+    board = await Board.populate(board, { path: 'issues' });
+    if (board) {
+      const { lifecycles, issues } = board;
+      const responseObject = lifecycles.reduce(
+        (acc, lc) => Object.assign(acc, { [lc]: { issues: [] } }),
+        {},
+      );
+      issues.forEach((issue) => {
+        const { lifeCycle } = issue;
+        responseObject[lifeCycle].issues.push(issue);
+      });
+      res.status(200).send(buildResponse(true, '', { lifeCycles: responseObject }));
+    }
+  } catch (err) {
+    res.status(500).send(buildResponse(false, `Error , ${err}`));
```

Comment:
Don't append error messages with `Error`,

```suggestion
    res.status(500).send(buildResponse(false, err));
```

Also, this message will be shown to the user, So describe the message relative to the user's understanding.

I suggest don't send system messages as in response.
Instead, add a common util function that will alert the developer.